### DISCUSSION TITLE
Hibernate Reactive Transactions with Reactive REST Client

### DISCUSSION THREAD

**Discussion by nenad-kosanovic-tam** [micronaut-projects/micronaut-data#2083] [Q&A] [answered]
When using Hibernate Reactive with vertx-mysql-client and have service method annotated with transaction, vert.x thread is used.
If there are no other async calls, except to Hibernate Reactive repository, all works good.

But, if in this execution chane, we need to call Reactive REST client, this call will switch to different thread (not vert.x), and exception will be thrown on callback, as new session will be created instead of re-using the previously existing session residing in context.

`HR000069: Detected use of the reactive Session from a different Thread than the one which was used to open the reactive Session - this suggests an invalid integration; original thread [52]: 'vert.x-eventloop-thread-1' current Thread [55]: 'vert.x-eventloop-thread-2`

How should we work with Reactive REST Client and Hibernate Reactive Transactions?

Here is the repository and test method in it, that will reproduce issue: https://github.com/nenad-kosanovic-tam/Hibernate-Reactive-Transactions-with-Reactive-REST-Client

Issue is reproducable by running test ThingServiceTest#createWithRestCall_should_fail.

**Comment by graemerocher** [comment]
please provide an example that reproduces the issue

**Reply by nenad-kosanovic-tam** [reply]
@graemerocher  Here is the repository and test method in it, that will reproduce issue: https://github.com/nenad-kosanovic-tam/Hibernate-Reactive-Transactions-with-Reactive-REST-Client

Issue is reproducable by running test ThingServiceTest#createWithRestCall_should_fail.

**Comment by nenad-kosanovic-tam** [comment]
Please let me know if you need any more details on this topic?

**Comment by nenad-kosanovic-tam** [comment]
Any update here?

**Reply by dstepanov** [reply, marked as answer]
It's a problem in Hibernate Reactive https://github.com/hibernate/hibernate-reactive/issues/1667
